crates/e2e/tests/e2e/jit_orders.rs

```diff
@@ -0,0 +1,226 @@
+        solver.account(),
+        onchain
+            .contracts()
+            .uniswap_v2_factory
+            .create_pair(token_a.address(), token_b.address())
+    );
+    tx!(
+        solver.account(),
+        token_a.approve(
+            onchain.contracts().uniswap_v2_router.address(),
+            to_wei(1000)
+        )
+    );
+    tx!(
+        solver.account(),
+        token_b.approve(
+            onchain.contracts().uniswap_v2_router.address(),
+            to_wei(1000)
+        )
+    );
+    tx!(
+        solver.account(),
+        onchain.contracts().uniswap_v2_router.add_liquidity(
+            token_a.address(),
+            token_b.address(),
+            to_wei(100),
+            to_wei(100),
+            0_u64.into(),
+            0_u64.into(),
+            solver.address(),
+            U256::max_value(),
+        )
+    );
+
+    // Approve GPv2 for trading
+    tx!(
+        trader_a.account(),
+        token_a.approve(onchain.contracts().allowance, to_wei(100))
+    );
+    tx!(
+        trader_a.account(),
+        token_b.approve(onchain.contracts().allowance, to_wei(100))
+    );
+    tx!(
+        solver.account(),
+        token_a.approve(onchain.contracts().allowance, to_wei(100))
+    );
+    tx!(
+        solver.account(),
+        token_b.approve(onchain.contracts().allowance, to_wei(100))
+    );
+
+    // Place Orders
+    let services = Services::new(onchain.contracts()).await;
+
+    let solution = Solution {
+        id: 0,
+        prices: HashMap::from([
+            (token_a.address(), to_wei(1)),
+            (token_b.address(), to_wei(1)),
+        ]),
+        trades: vec![
+            solvers_dto::solution::Trade::Jit(solvers_dto::solution::JitTrade {
+                order: JitOrder {
+                    owner: trader_a.address(),
+                    sell: Asset {
+                        amount: to_wei(10),
+                        token: token_b.address(),
+                    },
+                    buy: Asset {
+                        amount: to_wei(1),
+                        token: token_a.address(),
+                    },
+                    kind: OrderKind::Sell,
+                    partially_fillable: false,
+                    valid_to: model::time::now_in_epoch_seconds() + 300,
+                    app_data: Default::default(),
+                    receiver: solver.address(),
+                }
+                .sign(
+                    EcdsaSigningScheme::Eip712,
+                    &onchain.contracts().domain_separator,
+                    SecretKeyRef::from(&SecretKey::from_slice(solver.private_key()).unwrap()),
+                ),
+                executed_amount: to_wei(10),
+            }),
+            solvers_dto::solution::Trade::Fulfillment(solvers_dto::solution::Fulfillment {
+                executed_amount: to_wei(10),
+                fee: Some(0.into()),
+                // Dummy as it will be overwritten in the solver
+                order: [0; 56],
```

Comment:
I think instead of having logic inside the solver to overwrite dummy data in the test we should first create the user order with the orderbook which returns the actual order uid and then write that into the solution we want to return.

Reply:
> A way to solve this issue would be implementing some type of mspc

Had this PR marinating in my head and came back to suggest exactly this. That way you can super easily set up the solver and adjust as is needed. This also allows the same mock solver to return multiple different solutions throughout the test which I like! I just would use a `Mutex` instead of a `channel` to make it as simple as possible as a channel would introduce edge cases like:
1. how do I handle a tester sending lots and lots of solutions into the channel
2. should I return every solution exactly once or reuse it until a new solution appears
3. probably more that I currently don't think about

With a mutex it will be very clear what should happen just from reading the test case.